selfdrive/sensord/sensors_qcom2.cc

```diff
@@ -67,37 +69,19 @@ void interrupt_loop(int fd, std::vector<Sensor *>& sensors, PubMaster& pm) {
     int num_events = err / sizeof(*evdata);
     uint64_t ts = evdata[num_events - 1].timestamp - offset;
 
-    MessageBuilder msg;
-    auto orphanage = msg.getOrphanage();
-    std::vector<capnp::Orphan<cereal::SensorEventData>> collected_events;
-    collected_events.reserve(sensors.size());
-
     for (Sensor *sensor : sensors) {
-      auto orphan = orphanage.newOrphan<cereal::SensorEventData>();
-      auto event = orphan.get();
-      if (!sensor->get_event(event)) {
+      MessageBuilder msg;
+      std::string service;
+      if (!sensor->get_event(msg, service, ts)) {
```

Comment:
the sensors classes shouldn't know anything about services. I'd make this an std::map, something like:
```
std::map sensors = {
  lsmGyroClass, "gyroscope",
  bmxGyroClass, "gyroscope2",
};
```

Reply:
yea, agree on that, thought about that as well, but thought then everything is more contained in the sensor